src/algebra/pointwise.lean

```diff
@@ -570,6 +714,98 @@ instance is_central_scalar [has_scalar α β] [has_scalar αᵐᵒᵖ β] [is_ce
   is_central_scalar α (set β) :=
 ⟨λ a S, congr_arg (λ f, f '' S) $ by exact funext (λ _, op_smul_eq_smul _ _)⟩
 
+end smul
+
+section vsub
+variables {ι : Sort*} {κ : ι → Sort*} [has_vsub α β] {s s₁ s₂ t t₁ t₂ : set β} {a : α}
+  {b c : β}
+include α
```

Comment:
I'm surprised this is necessary!

Reply:
Is it all that surprising? `vsub` goes `β → β → α`, not `β → α → α`. Lean can't infer `α` from context.

Reply:
Ah I missed that not each lemma has `α` in its signature from other variables already